src/core/js/views/questionView.js

```diff
@@ -279,6 +282,38 @@ define([
         // be a full reset
         resetQuestion: function() {},
 
+        determineButtonState: function() {
```

Comment:
Just my two cents...the easiest way I find reading the logic is with with a separate getter, then you can return early:
```js
getButtonState: function() {
    if (this.get('_isCorrect')) return 'correct';

    if (this.get('_attemptsLeft') === 0) {
        return this.get('_canShowModelAnswer') ? 'showCorrectAnswer' : 'incorrect';
    }

    return this.get('_isSubmitted') ? 'reset' : 'submit';
}
```
P.S. **not tested** – just the theory.